pkg/flavor/flavor.go

```diff
@@ -72,9 +73,13 @@ func (m *methods) FindProfile(vm *virtv1.VirtualMachine) (*flavorv1alpha1.Virtua
 	}
 }
 
-func (m *methods) ApplyToVmi(field *k8sfield.Path, profile *flavorv1alpha1.VirtualMachineFlavorProfile, vmiSpec *virtv1.VirtualMachineInstanceSpec) Conflicts {
+func (m *methods) ApplyToVmi(field *k8sfield.Path, profile *flavorv1alpha1.VirtualMachineFlavorProfile, vmiSpec *virtv1.VirtualMachineInstanceSpec, vmiMeta *metav1.ObjectMeta) Conflicts {
 	var conflicts Conflicts
 
+	if vmiMeta.Annotations == nil {
+		vmiMeta.Annotations = make(map[string]string)
+	}
+	vmiMeta.Annotations["Flavor"] = profile.Name
```

Comment:
Two things here,

1. this is the flavor profile's name, not the flavor's name.
2. to ensure we don't stomp on user create annoations, we need to use the kubevirt.io prefix. so something like `kubevirt.io/flavor-name` and `kubevirt.io/cluster-flavor-name`. These shuld be defined types.go with our other labels and annotations